File: backend/routes/orchid_workflow.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func orchidWorkflowRoutes(rg *gin.RouterGroup) {
	rg.GET("/orchid-workflows", handlers.GetOrchidWorkflowDefinitions)
	rg.GET("/orchid-workflows/:id", handlers.GetOrchidWorkflowDefinition)
	rg.POST("/orchid-workflows", handlers.CreateOrchidWorkflowDefinition)
	rg.PUT("/orchid-workflows/:id", handlers.UpdateOrchidWorkflowDefinition)
	rg.DELETE("/orchid-workflows/:id", handlers.DeleteOrchidWorkflowDefinition)
	rg.GET("/orchid-workflow-histories", handlers.GetOrchidWorkflowHistories)
	rg.POST("/orchid-workflow-transfer", handlers.TransferOrchidWorkflowTask)
	rg.POST("/orchid-workflow-skip", handlers.SkipOrchidWorkflowNode)
	rg.POST("/orchid-workflow-seed", handlers.SeedOrchidWorkflowTemplates)
}

File: frontend/src/api/orchid_workflow.js
import http from './http'

export const getOrchidWorkflows = (params) => http.get('/orchid-workflows', { params })
export const getOrchidWorkflowDefinition = (id) => http.get(`/orchid-workflows/${id}`)
export const createOrchidWorkflow = (data) => http.post('/orchid-workflows', data)
export const updateOrchidWorkflow = (id, data) => http.put(`/orchid-workflows/${id}`, data)
export const deleteOrchidWorkflow = (id) => http.delete(`/orchid-workflows/${id}`)

export const getOrchidWorkflowHistories = (params) => http.get('/orchid-workflow-histories', { params })
export const getMyPendingApprovals = (params) => http.get('/approvals/pending', { params })
export const transferOrchidWorkflowTask = (params, data) => http.post('/orchid-workflow-transfer', data, { params })
export const skipOrchidWorkflowNode = (params, data) => http.post('/orchid-workflow-skip', data, { params })
export const seedOrchidWorkflowTemplates = () => http.post('/orchid-workflow-seed')
